modules/microshift-storage-device-classes.adoc

```diff
@@ -0,0 +1,47 @@
+// Module included in the following assemblies:
+//
+// * microshift_storage/volume-snapshots-microshift.adoc
+
+:_content-type: CONCEPT
+[id="microshift-storage-device-classes_{context}"]
+== Device classes
+
+You can create custom device classes by adding a `device-classes` array to your LVMS configuration. Add the array to the `/etc/microshift/lvmd.yaml` configuration file. A single device class must be set as the default. You must restart {product-title} for configuration changes to take effect.
+
+[NOTE]
+====
+Removing a device class while there are still persistent volumes or `VolumeSnapshotContent` objects connected to that device class breaks both thick and thin provisioning.
+====
+
+Multiple device classes can be defined in the `device-classes` array. These classes can be a mix of thick and thin volume configurations.
+
+.Example of a mixed `device class` array
+
+[source, terminal]
+----
+socket-name: /run/topolvm/lvmd.sock
+device-classes:
+  - name: ssd
+    volume-group: ssd-vg
+    spare-gb: 0 <1>
+    default: true
+  - name: hdd
+    volume-group: hdd-vg
+    spare-gb: 0 <1>
+  - name: thin
+    spare-gb: 0 <1>
+    thin-pool:
+      name: thin
+      overprovision-ratio: 10
+    type: thin
+    volume-group: ssd
+  - name: striped
+    volume-group: multi-pv-vg
+    spare-gb: 0 <1>
+    stripe: 2
+    stripe-size: "64"
+    lvcreate-options:<2>
+----
+[.small]
+<1> Setting spare-gb to anything other than `0` usually results in more space being allocated than expected.
```

Comment:
Using the same number for multiple callouts breaks on the Customer Portal. Unfortunately, we need to number sequentially and repeat the same text for each callout.

Reply:
Does it? That's news to me @michaelryanpeter. Do we have it in the guidelines anywhere?

I'd put `spare-gb` in backticks and give it a noun.

Also, is it really only "usually" or will it always allocate more space if you set it?

Could potentially reword to more actionable. First thoughts below, but probably could be better

```suggestion
<1> Set the `spare-gb` field to a value greater than `0` to allocate additional space.
```

Reply:
@bergerhoffer I think the guidelines could use an update, but here is a rejected Jira asking for support from DXP: https://issues.redhat.com/browse/CCS-6666

Reply:
An example in the wild: https://access.redhat.com/documentation/en-us/openshift_container_platform/4.12/html/installing/installing-on-aws#installation-aws-config-yaml_installing-aws-customizations

Reply:
Thanks for the example! I'm not surprised how bad that comes out with how ridiculous those conditionals are. I thought I had seen a basic repeater one work on the CP before, but that's hard to find one of them. Sounds like if it's not going to work then we should definitely get it into the guidelines.